### PR TITLE
Feature/test result enhancements

### PR DESCRIPTION
**Pull request by hariprasad-hp**
Enhance single test execution to auto-generate timestamped result JSON
--------------
Description
This commit introduces the ability to automatically generate uniquely named test result files for single test executions by including a timestamp in the filename.

Output path now includes timestamp for easy tracking (e.g., suite_name_single_20240527_145501.json)

If a directory is passed as output path, the system ensures it exists and appends a timestamped file

Prevents accidental overwriting of previous results

How Was This Tested?
------------
Executed with:
python3 main.py -v -b 127.0.0.1 -I device2 -c "SystemSettingsSetCec" -o ./test_result/
Confirmed that the result JSON file was created with the expected format and content.

Affected Files --> dab_tester.py

__________________________________
ci: add GitHub Action to auto-generate 8-digit test version on main push
-----------------

Description
This commit introduces a GitHub Actions workflow that automatically generates a new 8-digit test version string every time a push is made to the main branch. The version string is saved to test_version.txt.

Version format: YYYYMMDD + 2-digit counter (e.g., 2024052701)

Ensures every build has a unique and traceable version ID

test_version.txt can be read during test runs and injected into result JSON

Workflow File --> .github/workflows/update-test-version.yml

________________________________________
1. Enhanced JSON Test Output Format
--------
Introduced get_test_version() function to read test version from test_version.txt.

Each result JSON now includes metadata: version, suite_name, device_id, and execution_time.

Test result files are named with timestamps for uniqueness and traceability.
__________
fix: increase timeout for language setting due to reboot delay
   ---------
    - Updated 'system/settings/set' language operation timeout from 3000 ms to 120000 ms.
    - On Android devices, changing the system language often requires a reboot to take full effect in the UI.
    - Since reboot duration can take 1–2 minutes, this change allows the test to wait appropriately for system readiness.

### CODE REVIEW

File: .github/workflows/update-test-version.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  update-version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Generate 8-digit Test Version
+        run: |
+          DATE=$(date +'%y%m%d')
+          VERSION_FILE="test_version.txt"
```
Comment (bgy36ww) [inline .github/workflows/update-test-version.yml:18]: Can you add a test to cover if test_version.txt is missing?

File: .github/workflows/update-test-version.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Update Test Version on Main Push
```
Comment (bgy36ww) [inline .github/workflows/update-test-version.yml:1]: Two things to check after merge:
Make sure workflow bot is allowed to push changes
Check branch protection. I think that might cause an issue.

File: dab_tester.py [hunk 3]
```diff
@@ -28,50 +36,119 @@ def Execute_Test_Case(self, device_id, test_case):
...
+                "total": 1,
+                "passed": 0,
+                "failed": 0,
+                "skipped": 0
+            },
+            "test_result_list": []
+        }
+    
+        if not isinstance(test_case, tuple) or len(test_case) < 5:
+            raise ValueError("Invalid test_case structure. Expected tuple with 5 elements.")
```
Comment (bgy36ww) [inline dab_tester.py:82]: Is it possible to add a test to cover this edge case?
